File: TopInterview150/Stack/ValidParentheses/solution.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> stack;
        for (char c : s)
        {
            if (c == '[')
                stack.push(']');
            else if (c == '{')
                stack.push('}');
            else if (c == '(')
                stack.push(')');
            else if (stack.empty() || stack.top() != c)
                return false;
            else stack.pop();
        }
        return stack.empty();
    }
};

File: TopInterview150/Stack/ValidParentheses/solution.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack = []
        for c in s:
            if c == '(':
                stack.append(')')
            elif c == '{':
                stack.append('}')
            elif c == '[':
                stack.append(']')
            elif len(stack) == 0 or stack.pop() != c:
                return False
        return len(stack) == 0
    